search-commons/src/main/java/no/unit/nva/search/ElasticSearchHighLevelRestClient.java

```diff
@@ -152,23 +154,10 @@ public AcknowledgedResponse prepareIndexForBatchInsert() throws IOException {
                    : createNewIndexWithOptimizedSettings(indexSettings);
     }
 
-    public List<BulkResponse> batchInsert(List<IndexDocument> indexDocuments) {
-        List<List<IndexDocument>> bulks = Lists.partition(indexDocuments, BULK_SIZE);
-        return bulks.stream()
-            .parallel()
-            .map(attempt(this::insertBatch))
-            .map(Try::orElseThrow)
-            .collect(Collectors.toList());
-    }
-
-    private BulkResponse insertBatch(List<IndexDocument> bulk) throws IOException {
-        BulkRequest request = new BulkRequest();
-        for (IndexDocument document : bulk) {
-            request.add(getUpdateRequest(document));
-        }
-        request.setRefreshPolicy(RefreshPolicy.WAIT_UNTIL);
-        request.waitForActiveShards(ActiveShardCount.ONE);
-        return elasticSearchClient.bulk(request, RequestOptions.DEFAULT);
+    public Stream<BulkResponse> batchInsert(Stream<IndexDocument> indexDocuments) {
```

Comment:
replaced list with stream so that we do not have to load all index documents at once.